### PR TITLE
Tests are implemented

### PR DESCRIPTION
**Pull request by ovuruska**

### CODE REVIEW

File: deepinfra/models/base/automatic_speech_recognition.py [hunk 1]
```diff
@@ -26,4 +26,4 @@ def generate(self, body) -> AutomaticSpeechRecognitionResponse:
         response = self.client.post(
             form_data, {"headers": {"content-type": form_data.content_type}}
         )
-        return AutomaticSpeechRecognitionResponse(**response.json())
+        return response.json()
```
Comment (ichernev) [inline deepinfra/models/base/automatic_speech_recognition.py:29]: Why do you remove the conversion to the ASRResponse? Now the annotation is wrong (this is not C++, it won't auto-call the constructor or anything).

File: deepinfra/models/base/embeddings.py [hunk 2]
```diff
@@ -15,4 +15,4 @@ def generate(self, body) -> EmbeddingsResponse:
         :return:
         """
         response = self.client.post(body)
-        return EmbeddingsResponse(**response.json())
+        return response.json()
```
Comment (ichernev) [inline deepinfra/models/base/embeddings.py:18]: Same here.

File: deepinfra/models/base/text_generation.py [hunk 3]
```diff
@@ -23,4 +23,4 @@ def generate(self, body: dict) -> TextGenerationResponse:
         :return:
         """
         response = self.client.post(body)
-        return TextGenerationResponse(**response.json())
+        return response.json()
```
Comment (ichernev) [inline deepinfra/models/base/text_generation.py:26]: Same here

File: deepinfra/models/base/text_to_image.py [hunk 4]
```diff
@@ -16,4 +16,4 @@ def generate(self, input):
         """
         body = {"input": input}
         response = self.client.post(body)
-        return TextToImageResponse(**response.json())
+        return response.json()
```
Comment (ichernev) [inline deepinfra/models/base/text_to_image.py:19]: Same here. You also don't have annotation in the generate method.

You can use template-like specialization in inheritance, so the generate method won't complain it has a different signature, if that is the problem.

**Review by ichernev — COMMENTED** [review]
Having the right in/out type annotations in `generate` method is critical. Everything else has to be fleshed out around that, not the other way around.

File: deepinfra/models/base/automatic_speech_recognition.py (hunk 1 above)
Comment (ovuruska) [reply deepinfra/models/base/automatic_speech_recognition.py:29]: This code works inconsistenly before 3.10. It is related with kw_only keyword.
What about migrating our request and response schemas from dataclasses to pydantic?
Yes, it does come up with a dependency, but, client library will work more robustly across different Python versions.

**Pull request merged**